lib/utils/remote-build.ts

```diff
@@ -43,17 +44,25 @@ export interface RemoteBuild {
 
 	// For internal use
 	releaseId?: number;
+	hadError?: boolean;
 }
 
 interface BuilderMessage {
 	message: string;
 	type?: string;
 	replace?: boolean;
+	isError?: boolean;
 	// These will be set when the type === 'metadata'
 	resource?: string;
 	value?: string;
 }
 
+export class RemoteBuildFailedError extends TypedError {
+	public constructor() {
```

Comment:
You can do:
```typescript
public constructor(message = 'Remote build failed') {
	super(message)
}
```
to keep the ability to specify custom messages

Reply:
Good idea, I like it.